prog/main.go

```diff
@@ -115,12 +117,14 @@ func main() {
 		flags         = flags{}
 		mode          string
 		debug         bool
+		weaveEnabled  bool
 		weaveHostname string
 	)
 
 	// Flags that apply to both probe and app
 	flag.StringVar(&mode, "mode", "help", "For internal use.")
 	flag.BoolVar(&debug, "debug", false, "Force debug logging.")
+	flag.BoolVar(&weaveEnabled, "weave", true, "Enable Weave Net integrations.")
```

Comment:
Boolean flags with a default value of true are hard for users to grok, and don't allow the "--flag" shorthand.

So I suggest flipping this to "noweave" (or "no-weave" or whatever conventions you follow).